src/proto/streams/recv.rs

```diff
@@ -368,19 +381,33 @@ impl Recv {
 
         // Ensure that there is enough capacity on the connection before acting
         // on the stream.
-        if self.flow.window_size() < sz || stream.recv_flow.window_size() < sz {
-            return Err(RecvError::Connection(Reason::FLOW_CONTROL_ERROR));
+        self.consume_connection_window(sz)?;
+
+        if is_ignoring_frame {
+            trace!("recv_data frame being ignored on locally reset {:?} for some time", stream.id);
+            return Ok(());
```

Comment:
So the connection level window is consumed right above, but I don't see where it is "released". Specifically, how connection level window is not leaked in this case.

It would be nice to have a test for this.

[Related](https://github.com/carllerche/h2/pull/174/files#r151241765)